src/Core/src/Handlers/Layout/LayoutHandler.Android.cs

```diff
@@ -119,7 +119,7 @@ void EnsureZIndexOrder(IView child)
 			}
 
 			AView platformChildView = child.ToPlatform(MauiContext!);
-			var currentIndex = IndexOf(PlatformView, platformChildView);
+			var currentIndex = PlatformView.IndexOfChild(platformChildView);
```

Comment:
What is `IndexOfChild()` that is used here? Is that a different method than `IndexOf()`?

Reply:
It is a method from Xamarin.Android which accomplishes the same thing using Java instead of C#

Reply:
Yes, calling `ViewGroup.IndexOfChild()`:

* https://developer.android.com/reference/android/view/ViewGroup#indexOfChild(android.view.View)

Should be much better than calling `ChildCount` and `GetChildAt()` N times. 👍